pkg/controller/certificates/trigger/policies/gatherer_test.go

```diff
@@ -0,0 +1,256 @@
+/*
+Copyright 2020 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package policies
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	corelisters "k8s.io/client-go/listers/core/v1"
+	"k8s.io/utils/pointer"
+
+	cmapi "github.com/jetstack/cert-manager/pkg/apis/certmanager/v1"
+	cmlisters "github.com/jetstack/cert-manager/pkg/client/listers/certmanager/v1"
+)
+
+func TestDataForCertificate(t *testing.T) {
+	tests := []struct {
+		name                       string
+		mockSecretLister           func(t *testing.T) secretListerMock
+		mockListRequests           func(t *testing.T) requestListerMock
+		givenCrt                   *cmapi.Certificate
+		wantCurrentRevisionRequest *cmapi.CertificateRequest
+		wantSecret                 *corev1.Secret
+		wantErr                    string
+	}{
+		{
+			name:             "the returned secret should stay nil when it is not found",
+			givenCrt:         &cmapi.Certificate{Spec: cmapi.CertificateSpec{SecretName: "secret-1"}, ObjectMeta: metav1.ObjectMeta{Name: "a"}},
```

Comment:
nit: can we make these struct definitions multi-line, its quite hard to read all on one line :smile:

Reply:
We can also make use of https://github.com/jetstack/cert-manager/blob/master/test/unit/gen/certificaterequest.go etc, which makes reading the input to these tests much easier to parse.